DBM-Party-Vanilla/Classic/SOD_DemonFallCanyon/Grimroot.lua

```diff
@@ -3,25 +3,113 @@ local L		= mod:GetLocalizedStrings()
+
+-- Gloom seems very predictable
+-- "Gloom-460727-npc:226923-000012D5C6 = pull:30.8, 30.8, 30.7",
+-- "Gloom-460727-npc:226923-000012D91C = pull:30.8, 30.8",
+-- Needs to be kicked, we did this once and got 30.7 instead of 30.8, I guess 30.7 timer is good enough
+
+-- Tender's rage isn't predictable (but maybe it's based on health?)
+-- "Tender's Rage-460703-npc:226923-000012D5C6 = pull:19.4, 44.8",
+-- "Tender's Rage-460703-npc:226923-000012D91C = pull:23.5, 28.0",
+-- "<87.01 21:47:15> [CLEU] SPELL_AURA_APPLIED#2632#Creature-0-5252-2784-26746-226923-000012D91C#Grimroot#Creature-0-5252-2784-26746-226923-000012D91C#Grimroot#460703#Tender's Rage#BUFF#nil#nil#nil#nil#nil",
+-- "<94.53 21:47:23> [CLEU] SPELL_AURA_REFRESH#2632#Creature-0-5252-2784-26746-226923-000012D91C#Grimroot#Creature-0-5252-2784-26746-226923-000012D91C#Grimroot#460703#Tender's Rage#BUFF#nil#nil#nil#nil#nil",
+-- "<102.52 21:47:31> [CLEU] SPELL_AURA_REMOVED#2632#Creature-0-5252-2784-26746-226923-000012D91C#Grimroot#Creature-0-5252-2784-26746-226923-000012D91C#Grimroot#460703#Tender's Rage#BUFF#nil#nil#nil#nil#nil",
+-- Can be refreshed :o
+-- It's a "Frenzy" effect so we can probably dispel it with Tranquilizing Shot once MC is out, fun mechanic
+
+
 mod:RegisterEventsInCombat(
---	"SPELL_CAST_START",
---	"SPELL_CAST_SUCCESS,
---	"SPELL_AURA_APPLIED"
+	"SPELL_DAMAGE 460512",
+	"SPELL_PERIODIC_DAMAGE 460512",
+	"SPELL_MISSED 460512",
+	"SPELL_PERIODIC_MISSED 460512",
+	"SPELL_CAST_SUCCESS 460509",
+	"SPELL_CAST_START 460727",
+	"SPELL_AURA_APPLIED 460703",
+	"SPELL_AURA_REFRESH 460703",
+	"SPELL_AURA_REMOVED 460703"
 )
 
---function mod:OnCombatStart(delay)
+local specWarnGTFO		= mod:NewSpecialWarningGTFO(460512, nil, nil, nil, 1, 8)
+local specWarnGloom		= mod:NewSpecialWarningInterrupt(460727, "HasInterrupt", nil, nil, 1, 2)
+local specWarnFrenzy	= mod:NewSpecialWarningDispel(460703, "RemoveEnrage", nil, nil, 1, 2)
+
+local yellTears			= mod:NewIconRepeatYell(460512)
+
+local timerGloom		= mod:NewCastTimer(30.7, 460727)
+local timerFrenzy		= mod:NewBuffActiveTimer(8, 460703)
 
---end
+local playerGuid = UnitGUID("player")
+local playerName = UnitName("player")
 
+function mod:OnCombatStart(delay)
+	timerGloom:Start(30.8 - delay)
+end
+
+function mod:SPELL_DAMAGE(_, _, _, _, destGUID, _, _, _, spellId, spellName)
+	if spellId == 460512 and destGUID == playerGuid and self:AntiSpam(2.5, 1) then -- Spam every 3 ticks
+		specWarnGTFO:Show(spellName)
+		specWarnGTFO:Play("watchfeet")
+	end
+end
+mod.SPELL_MISSED = mod.SPELL_DAMAGE
+mod.SPELL_PERIODIC_DAMAGE = mod.SPELL_DAMAGE
+mod.SPELL_PERIODIC_MISSED = mod.SPELL_DAMAGE
+
+-- TODO: i'm not sure if this is correct, the log above has the SPELL_AURA_APPLIED on a different player than the target
+-- But the target was me, and I probably dodged, so worth trying.
+function mod:CorruptedTearsTarget(target)
+	if target == playerName then
+		yellTears:Yell(8)
```

Comment:
The Icon Repeat yell object is more option description but it does not AUTOMATICALLY repeat (at least at this time). this would yell once and never repeat without a local scheduling loop.

Reply:
which is fine here, i just wanted a lightweight yell without the full spell name. will keep as is for now, might want to add another option like "just show 💀💀💀💀" tomorrow